### DISCUSSION TITLE
What is the best practice when doing multiple filtering: two filters vs. one filter?

### DISCUSSION THREAD

**Discussion by tanvirstreame** [sqlalchemy/sqlalchemy#9711] [Usage Questions]
While doing multiple filtering, some people use separate filters for separate conditions -

```
db.session.query(Book)
      .filter(Segment.guid == guid)
      .filter(Book.deleted == False)
```

Others do it with a single filter -

```
db.session.query(Book).filter(
         Segment.guid == guid,
         Book.deleted == False
 )
```

Q1. Which should be followed as best practice?
Q2. Is there any performance issue?

**Comment by CaselIT** [comment]
Hi,

The two do exactly the same thing, both resulting in `WHERE Segment.guid == guid AND Book.deleted == False`

> Q1. Which should be followed as best practice?

Any is fine. I don't think there is any best practice suggested in this case by SQLAlchemy

> Q2. Is there any performance issue?

The SQL generated is exactly the same. Regarding time to construct the statement that should be pretty much itentical
